src/world/FieldComputations.cpp

```diff
@@ -38,126 +35,38 @@ double FieldComputations::getPercentageOfGoalVisibleFromPoint(const Field &field
     return fmax(100 - blockadeLength / goalWidth * 100, 0.0);
 }
 
-std::vector<Line> FieldComputations::getBlockadesMappedToGoal(const Field &field, bool ourGoal, const Vector2 &point, std::vector<world_new::view::RobotView> robots, int id, bool ourTeam) {
-    const double robotRadius = Constants::ROBOT_RADIUS() + Constants::BALL_RADIUS();
-
-    Vector2 lowerGoalSide, upperGoalSide;
-    auto sides = getGoalSides(field, ourGoal);
-    lowerGoalSide = sides.start;
-    upperGoalSide = sides.end;
-
-    std::vector<Line> blockades = {};
-
-    // all the obstacles should be robots
-    for (auto const &robot : robots) {
-        if (robot->getId() == id && robot->getTeam() == (ourTeam ? world_new::Team::us : world_new::Team::them)) continue;
-        auto pos = robot->getPos();
-        double lenToBot = (point - pos).length();
-        // discard already all robots that are not at all between the goal and point, or if a robot is standing on this point
-        bool isRobotItself = lenToBot <= robotRadius;
-        bool isInPotentialBlockingZone = ourGoal ? pos.x < point.x + robotRadius : pos.x > point.x - robotRadius;
-        if (!isRobotItself && isInPotentialBlockingZone) {
-            // get the left and right sides of the robot
-            double theta = asin(robotRadius / lenToBot);
-            double length = sqrt(lenToBot * lenToBot - robotRadius * robotRadius);
-            Vector2 lowerSideOfRobot = point + Vector2(length, 0).rotate((Vector2(pos) - point).angle() - theta);
-            Vector2 upperSideOfRobot = point + Vector2(length, 0).rotate((Vector2(pos) - point).angle() + theta);
-            // map points onto goal line
-
-            // the forwardIntersection returns a double which is the scale of the vector projection
-            // this returns -1.0 if there is no intersections in the forward direction
-            double point1val = util::twoLineForwardIntersection(point, lowerSideOfRobot, lowerGoalSide, upperGoalSide);
-            double point2val = util::twoLineForwardIntersection(point, upperSideOfRobot, lowerGoalSide, upperGoalSide);
-            // Here is how we calculate the actual intersections using the above values
-            Vector2 point1 = point + (lowerSideOfRobot - point) * point1val;
-            Vector2 point2 = point + (upperSideOfRobot - point) * point2val;
-            // we can use the
+std::vector<Line> FieldComputations::getVisiblePartsOfGoal(const Field &field, bool ourGoal, const Vector2 &point, world_new::view::WorldDataView &world) {
```

Comment:
```suggestion
std::vector<Line> FieldComputations::getVisiblePartsOfGoal(const Field &field, bool ourGoal, const Vector2 &point, world_new::view::WorldDataView world) {
```

Reply:
Why do you want to copy a class and not pass it by reference?